packages/nvidia_nat_mcp/src/nat/plugins/mcp/tool.py

```diff
@@ -85,17 +85,21 @@ def _convert_from_str(input_str: str) -> tool.input_schema:
         return tool.input_schema.model_validate_json(input_str)
 
     async def _response_fn(tool_input: BaseModel | None = None, **kwargs) -> str:
-        # Run the tool, catching any errors and sending to agent for correction
-        try:
-            if tool_input:
-                args = tool_input.model_dump()
-                return await tool.acall(args)
-
-            _ = tool.input_schema.model_validate(kwargs)
-            return await tool.acall(kwargs)
-        except Exception as e:
-            logger.warning("Error calling tool %s", tool.name, exc_info=True)
-            return str(e)
+         # Run the tool, catching any errors and sending to agent for correction
+         try:
+             if tool_input:
+                 # Exclude None values for optional parameters.
+                 # MCP server will use server-side defaults for omitted optional fields.
+                 args = tool_input.model_dump(exclude_none=True)
```

Comment:
This fix makes sense and has already been applied to the mcp_client function group. It wasn't added to mcp_tool wrapper as this function has been deprecated and is no longer maintained.